src/applications/vaos/new-appointment/newAppointmentFlow.js

```diff
@@ -61,17 +61,26 @@ function getFacilityPageKey(state) {
 }
 
 async function vaFacilityNext(state, dispatch) {
-  let eligibility;
+  let eligibility = getEligibilityStatus(state);
 
   if (selectUseFlatFacilityPage(state)) {
-    const facility = getChosenFacilityInfo(state);
-    const siteId = getSiteIdFromFakeFHIRId(facility.id);
-    eligibility = await dispatch(checkEligibility(facility, siteId));
+    // Fetch eligibility if we haven't already
+    if (eligibility.direct === null && eligibility.request === null) {
+      const location = getChosenFacilityInfo(state);
+      const siteId = getSiteIdFromFakeFHIRId(location.id);
+
+      eligibility = await dispatch(
+        checkEligibility({
+          location,
+          siteId,
+          showModal: true,
+        }),
+      );
+    }
+
     if (!eligibility.direct && !eligibility.request) {
       return VA_FACILITY_V2_KEY;
     }
-  } else {
```

Comment:
Why did this else go away? It looks like it's for users on the V1 page.

Reply:
This gets set above on line 64 of the updated version